polkadot/node/core/pvf/src/prepare/worker_intf.rs

```diff
@@ -219,6 +219,20 @@ async fn handle_response(
 		return Outcome::TimedOut
 	}
 
+	let checksum = match tokio::fs::read(&tmp_file).await {
+		Ok(bytes) => blake3::hash(&bytes),
+		// FIXME eagr: handle error properly
+		Err(_) => return Outcome::Unreachable,
+	};
+
+	let file_name_partial = artifact_path_partial
+		.file_name()
+		.expect("the path should never terminate in `..`");
+	let mut file_name = file_name_partial.to_os_string();
+	file_name.push("_0x");
+	file_name.push(checksum.to_hex().as_str());
+	let artifact_path = artifact_path_partial.with_file_name(&file_name);
```

Comment:
We should have a helper function for this and some unit tests. In the tests we can call `from_file_name` on the result of the helper function to verify that we get expected values.